best_line.h

```diff
@@ -0,0 +1,52 @@
+#ifndef BEST_LINE_H
+#define BEST_LINE_H
+
+#include <float.h>
+#include <limits.h>
+
+#include <iostream>
+#include <unordered_map>
+#include <vector>
+
+using namespace std;
+
+/*Object to represent a point with x,y coordinates*/
+struct Point {
+  double x;
+  double y;
+
+  Point(double a, double b) : x(a), y(b) {}
+};
+
+/*Class for all info necessary to describe a line.
+NOTE: It is assumed that if line is vertical or horizontal, intercept and
+x_intercept respectively will not be considered.*/
+class Line {
+ public:
```

Comment:
this doesn't look like a class, since all data members are public. Make it struct, drop public. 
Or, if there is a reason to, make it a class, and make all data members private.

Reply:
In this case, I would probably make it a class.

Reply:
style: public, then private